src/Symfony/Component/PropertyInfo/Tests/Extractor/PhpDocExtractorTest.php

```diff
@@ -887,6 +888,23 @@ public static function constructorTypesProvider(): iterable
         yield ['mixed', Type::mixed()];
     }
 
+    /**
+     * @dataProvider constructorPromotedPropertyWithDocblockProvider
+     */
+    public function testExtractConstructorPromotedPropertyWithDocblock(string $property, ?Type $type)
+    {
+        $this->assertEquals($type, $this->extractor->getTypeFromConstructor(Php80PromotedDummyWithDocblock::class, $property));
+    }
+
+    /**
+     * @return iterable<array{0: string, 1: ?Type}>
+     */
+    public static function constructorPromotedPropertyWithDocblockProvider(): iterable
+    {
+        // Test that promoted properties with @var docblock are recognized
+        yield ['dates', Type::list(Type::object(\DateTimeImmutable::class))];
```

Comment:
Can you also test with the `@param` here? 

And maybe with both `@var` and `@param` to see which one has the priority?
It could be great if the priority is the same as the `PhpDocAwareReflectionTypeResolver`, which seems to give priority to `@var`.

Reply:
@mtarld all right ! I added some tests and gave priority to `@var` to be consistent with `PhpDocAwareReflectionTypeResolver`